test/sshkit/ssh_test.exs

```diff
@@ -59,6 +60,30 @@ defmodule SSHKit.SSHTest do
       refute_received :closed_sandbox_connection
     end
 
+    test "its logs the connection when dry_run is enabled" do
```

Comment:
If not written like this for a specific reason, I'd suggest to align the naming of tests with the rest of `SSHKit`:
```elixir
test "log the connection when dry_run is enabled" do
  # do stuff
end
```

In my opinion this also better reflects the way one would address and talk about modules and their functions: it's not really testing the module's behavior in the way one would test a class in Ruby. I'd say one cannot really speak of a module having a *behavior*  at all - it's simply a collection of pure functions 🗃 

*(this of course relates to all tests in this file)*